packages/core/src/electron-main/electron-main-application.ts

```diff
@@ -348,31 +373,37 @@ export class ElectronMainApplication {
      * Save the window geometry state on every change.
      */
     protected attachSaveWindowState(electronWindow: BrowserWindow): void {
-        const saveWindowState = () => {
-            try {
-                const bounds = electronWindow.getBounds();
-                this.electronStore.set('windowstate', {
-                    isFullScreen: electronWindow.isFullScreen(),
-                    isMaximized: electronWindow.isMaximized(),
-                    width: bounds.width,
-                    height: bounds.height,
-                    x: bounds.x,
-                    y: bounds.y
-                });
-            } catch (e) {
-                console.error('Error while saving window state:', e);
-            }
-        };
         let delayedSaveTimeout: NodeJS.Timer | undefined;
         const saveWindowStateDelayed = () => {
             if (delayedSaveTimeout) {
                 clearTimeout(delayedSaveTimeout);
             }
-            delayedSaveTimeout = setTimeout(saveWindowState, 1000);
+            delayedSaveTimeout = setTimeout(() => this.saveWindowState(electronWindow), 1000);
         };
-        electronWindow.on('close', saveWindowState);
+        electronWindow.on('close', () => {
+            this.saveWindowState(electronWindow);
+            this.originalFrames.delete(electronWindow.id);
+        });
         electronWindow.on('resize', saveWindowStateDelayed);
         electronWindow.on('move', saveWindowStateDelayed);
+        this.originalFrames.set(electronWindow.id, this.frame);
+    }
+
+    protected saveWindowState(electronWindow: BrowserWindow): void {
+        try {
+            const bounds = electronWindow.getBounds();
```

Comment:
I did run into an error during one of my first runs through, but I couldn't reproduce it:
```
Error while saving window state: TypeError: Cannot read property 'getBounds' of null
```

Reply:
It seems like `BrowserWindow.fromId()` can sometimes return `null`. I added a guard for that 👍